lib/tasks/whitehall_importer.rb

```diff
@@ -182,6 +184,12 @@ def published_author(whitehall_edition)
       whitehall_edition["revision_history"].select { |h| h["state"] == "published" }.first
     end
 
+    def valid_state?(whitehall_edition)
+      return false if whitehall_edition["state"] == "withdrawn" && !whitehall_edition["unpublishing"]
```

Comment:
Shouldn't this just be `whitehall_edition["state"] == "withdrawn" && whitehall_edition["unpublishing"]` as a withdrawn edition should always have an unpublishing object? 

The logic here currently does this I think (took a while to wrap my head around 😆):
* Return false if withdrawn && non_unpublishing
* Return true if not_withdrawn && unpublishing || not_withdrawn && non_unpublishing || withdrawn && unpublishing

If this is correct it could be wrote like this: `!(whitehall_edition["state"] == "withdrawn" && !whitehall_edition["unpublishing"])`

Reply:
This test will be used for other states too, e.g. "published", which should just fall through to the `true`, hence the slightly odd logic.